harness/compareArray.js

```diff
@@ -35,8 +35,11 @@ assert.compareArray = function(actual, expected, message) {
   assert(actual != null, `First argument shouldn't be nullish. ${message}`);
   assert(expected != null, `Second argument shouldn't be nullish. ${message}`);
   var format = compareArray.format;
-  assert(
-    compareArray(actual, expected),
-    `Expected ${format(actual)} and ${format(expected)} to have the same contents. ${message}`
-  );
+  var result = compareArray(actual, expected);
+
+  // The following prevents actual and expected from being iterated and evaluated
+  // more than once unless absolutely necessary.
+  if (!result) {
+    assert(false, `Expected ${format(actual)} and ${format(expected)} to have the same contents. ${message}`);
+  }
```

Comment:
@jugglinmike Rick and I discussed about this assertion message that is now deferred to execute only if compareArray fails. That might _avoid_ conflicts for existing tests migrating from pure compareArray to assert.compareArray.